### DISCUSSION TITLE
windows 11 server + mac os x ventura client + connected but control is not moving over

### DISCUSSION THREAD

**Discussion by uresh-kuruhuri** [debauchee/barrier#1876] [Q&A]
Setup:
Windows 11 (21H2 and updated to 22H2): Server
versions tried: v2.4.0

Mac OS Ventura: Client
versions tried: v2.4.0

I have been using barrier since few years in Win 10 and Ubuntu (18.0 through 22.04) as Server and Mac OS clients. It's been working fine. 

I got this new laptop with Win 11 and installed Barrier v2.4.0 and set it up as Server. With v2.4.0 client on Mac OS X it connected (after adding the relevant entries barrier.exe and barriers.exe in the Firewall allow list). However, the mouse pointer always stayed on the right bottom corner of the client and it never moves. However, keyboard was working. 

Since the mouse cannot be moved, the control cannot move back to server. Also, no scroll lock on mac (client) keyboard and scroll lock is OFF on the external keyboard connected to server (otherwise the control could not move over to client in the first place).

So, with v2.4.0 it is not working properly. 

Can anyone help with any pointers that can get the barrier work on Win 11 as Server and Mac OS as clients?

Thanks.

**Comment by uresh-kuruhuri** [comment]
This is what I found.

The default display settings in the Win 11 laptop came with **150% scaling** on the resolution. I changed it to 125% as in the picture below. See the highlighted area in red rectangle and yellow text.

![Screenshot 2023-01-12 135506](https://user-images.githubusercontent.com/1688530/212156701-6de345a2-2cbc-40b9-bc28-5bb2821c72ca.png)

However, when I reset the scaling to 100% with the highest resolution in barrier logs, it came up as below and the mouse moved back and forth with the server/client.
`[2023-01-12T13:46:45] DEBUG: screen shape: 0,-1080 2560x2680 (multi-monitor)`

Whenever I change the scaling other than 100 (e.g. to 125%, came up as below in barrier log), the above problem in the original post occurs.
`[2023-01-12T13:46:58] DEBUG: screen shape: 0,-1080 2181x2360 (multi-monitor)`

Also, on the Mac OS Ventura (client), changing the display resolution does not affect the operation or the original problem.

**Reply by uresh-kuruhuri** [reply]
why barrier would not work when the resolution is changed?

**Reply by uresh-kuruhuri** [reply]
Also, [changing this setting (as in the comment for issue#94)](https://github.com/debauchee/barrier/issues/94#issuecomment-979408763)  for the barriers.exe in Win 11 Server, helped in getting it worked finally with the resolution changes.

**Comment by ghost** [comment]
I have been using Barrier between a Window 10 pc and a Mac OS mac mini for years now with no issues (a few different versions of Mac OS, but now Ventura). The monitor attached to the mac has been a 4k screen, and the pc had multiple (1, 2, 3, or 4, depending on my life), all of which were standard 1920x1080 or some no scaled similar resolution.

I ran into the exact problem you described today, after switching the PC monitors to new 4k screen, which Windows 10 automatically scales to it's recommended 150%.

Barrier was still working and connected on both screens, but when moving the cursor, the log file indicated it saw the cursor move from the PC to the mac.. and that was the last entry, and I lost the use of the cursor on both screens. The only way to get it back that I found was to hit ctrl+alt+del on the pc (the barrier server in my setup, with the keyboard attached) that brought the cursor back to use on the pc, and the log file just indicated an interruption to barrier and the reconnection of the client.

Because it was only monitor hardware that had changed, I ended up searching here on the github discussion for 'resolution' and found this issue.

Changing the Windows 10 monitor resolutions to 100% instead of the recommended 150%, fixed the issue for me as well.

For what it was worth, I was running barrier 2.4.0 on the PC, and 2.3.3 on the mac. Upgrading the mac to 2.4.0 did not appear to have any effect on this issue.
